### PR TITLE
feat: update auth to v3

### PR DESCRIPTION
**Pull request by avneesh0612**

### CODE REVIEW

File: auth.config.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 import { ThirdwebAuth } from "@thirdweb-dev/auth/next";
 
 export const { ThirdwebAuthHandler, getUser } = ThirdwebAuth({
-  // Using environment variables to secure your private key is a security vulnerability.
-  // Learn how to store your private key securely:
-  // https://portal.thirdweb.com/sdk/set-up-the-sdk/securing-your-private-key
-  privateKey: process.env.ADMIN_PRIVATE_KEY as string,
-  domain: "example.com",
+  domain: process.env.NEXT_PUBLIC_AUTH_DOMAIN || "",
+  wallet: new PrivateKeyWallet(process.env.ADMIN_PRIVATE_KEY || ""),
```
Comment (adam-maj) [inline auth.config.ts]: Lets change the names of these environment variables to the ones we've been using elsewhere `NEXT_PUBLIC_THIRDWEB_AUTH_DOMAIN` and `THIRDWEB_AUTH_PRIVATE_KEY`

File: pages/_app.tsx [hunk 2]
```diff
@@ -10,8 +11,7 @@ function MyApp({ Component, pageProps }: AppProps) {
       desiredChainId={activeChainId}
       authConfig={{
         authUrl: "/api/auth",
-        domain: "example.com",
-        loginRedirect: "/",
+        domain: process.env.NEXT_PUBLIC_AUTH_DOMAIN as string,
```
Comment (adam-maj) [inline pages/_app.tsx]: Remember to change here as well

**Review by adam-maj — COMMENTED** [review]
Looks good after the env variable changes - I'm also assuming you've tested all these and they work.

**Comment by avneesh0612** [comment]
yea I have tested them

**Pull request merged**